### ISSUE TITLE
ESRCH on any command run from netlify dev (macos)

### ISSUE THREAD

**Issue by marcus13345** [netlify/cli#706]
**- Do you want to request a _feature_ or report a _bug_?**
Bug

**- What is the current behavior?**
```
❯ yarn netlify dev
yarn run v1.22.0
$ netlify dev
◈ Netlify Dev ◈
◈ Starting Netlify Dev with undefined
Waiting for localhost:8080.netlify.toml	node_modules	package.json	yarn-error.log	yarn.lock
◈ Error while killing child process: kill ESRCH
✨  Done in 0.90s.
```

**- If the current behavior is a bug, please provide the steps to reproduce.**
```
mkdir test
cd test
yarn init # do the prompts
yarn add netlify-cli
```
## `netlify.toml`
```
[dev]
  command = "ls"
```

## `package.json`
```
{
  ...
  "scripts": {
    "netlify": "netlify"
  }
  ...
}
```

Then run
```
yarn netlify dev
```

**- What is the expected behavior?**

netlify will list directories and serve nothing, but not break.

**- Local Environment Information**
```
❯ yarn netlify status --verbose
yarn run v1.22.0
$ netlify status --verbose
──────────────────────┐
 Current Netlify User │
──────────────────────┘
Name:  Marcus Gosselin
Email: marcus@mandworks.com
Teams:
  Trevor Gionet's team:   Collaborator
  Marcus Gosselin's team: Collaborator
 ›   Warning: Did you run `netlify link` yet?
 ›   Error: You don't appear to be in a folder that is linked to a site
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by marcus13345** [comment]
@RaeesBhatti would you mind elaborating on what was fixed here? Will this use case simply work now, should I be approaching it differently? 

Thanks

**Comment by RaeesBhatti** [comment]
@marcus13345 The problem was that when we didn't detect an application server, we didn't provide any environment variables to the "command" process, which meant that there was not $PATH and the OS couldn't find it. Which was a mistake that has been corrected and we've added tests to make sure environment variables are always passed.
This fix hasn't been released, I'll post an update here when we do release an update. Things should "Just Work" after an update.
